_docs/alerts.md

```diff
@@ -0,0 +1,69 @@
+# Alerts
+
+The basic `Alerts` library allows us to display notifications about the results of actions performed by the user.
+
+Since we use [htmx](https://htmx.org) in most places, we need to have a system that allows us to display inline 
+notifications. But also be able to display the notification after redirect.
+
+That's why this custom alerts library was created.
+
+## How is this working
+
+At the moment, alerts are built on top of the [Toast](https://daisyui.com/components/toast/) component.
+
+Alert types are set based on the available classes in the Toast component, but we should
+declare the type by omitting the "alert-" prefix. Thus, we have types: `success`, `error`, `info`.
+
+We use `AlertsFilter` class to determine whether we need to display alerts inline 
+(by using [hx-swap-oob](https://htmx.org/attributes/hx-swap-oob/)) or just by session's flash data 
+(when using redirect).
+
+### Setting alerts
+
+Setting an alert is quite easy:
+
+```php
+alerts()->set('success', 'Message goes here');
+
+alerts()->set('error', 'Message goes here');
+
+alerts()->set('info', 'Message goes here');
+```
+
+We can also have many alerts with the same type:
+
+```php
+alerts()->set('success', 'Message 1');
+
+alerts()->set('success', 'Message 2');
+````
+
+Also method chaining is available:
+
+```php
+alerts()->set('success', 'Message success')->set('error', 'Message error');
+```
+
+### Cleaning alerts
+
+We can clean alerts by type:
+
+```php
+alerts()->clean('success');
+```
+
+Or clear all the alerts at once:
+
+```php
+alerts()->clean();
+```
+
+### Changing position of alerts
+
+By default, alerts are displayed in the right bottom of the screen.
+To change that we have to edit view file: `app/Views/_alerts/container`.
+
+### Auto-hiding alerts
+
+By default, each alert is hidden after 5 seconds. We can change that by
+changing config variable: `Config\Forum::alertDisplayTime`.
```

Comment:
```suggestion
changing config variable: `Config\Forum::alertDisplayTime`.
```
Honestly, I believe, this value should be dynamically changeable. Because the length of the message text may be long in some cases, so the programmer should be able to set this amount of time for the long text.

e.g
 
```php
alerts()->set('success', 'Message success')->hidingAfter(3*1000);
```

Reply:
I don't remember where I saw it but somewhere recently they had a great solution: It had a thin "progress bar" on the bottom of the toast that would show how long it was lasting. When you hovered over it, or took the focus off of the page, the countdown would stop until you either came back to the page or the cursor moved out of it.

Reply:
I do think this is something that can be iterated over and shouldn't hold up this PR, though.